### PR TITLE
Log streaming

### PR DESCRIPTION
**Pull request by cjdenio**
This PR adds an endpoint to stream logs from an app's container.

### CODE REVIEW

File: dokku_plugin/subcommands/cid/cid.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	}
+
+	app_name := os.Args[2]
+
+	if err := common.VerifyAppName(app_name); err != nil {
+		fmt.Println("App not found")
+		os.Exit(1)
+	}
+
+	file, err := os.Open(path.Join(os.Getenv("DOKKU_ROOT"), app_name, "CONTAINER.web.1"))
```
Comment (anirudhb) [inline dokku_plugin/subcommands/cid/cid.go:26]: nit for future: figure out how to stream all container logs

File: pkg/api/apps/logs.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	dokku_conn := c.MustGet("dokkuconn").(*dokku.DokkuConn)
+	user := c.MustGet("user").(db.User)
+
+	app_id := c.Param("id")
+
+	var app db.App
+	result := db.DB.Raw(`SELECT apps.* FROM apps
+	JOIN teams ON teams.id = apps.team_id
+	JOIN team_users ON team_users.team_id = teams.id
+	WHERE apps.id = ? AND team_users.user_id = ?`, app_id, user.ID).Scan(&app)
```
Comment (anirudhb) [inline pkg/api/apps/logs.go]: Rewrite to use gorm's Joins function
Something like
```go
result := db.DB.Joins("teams ON teams.id = apps.team_id").
    Joins("team_users ON team_users.team_id = teams.id").
    Where("apps.id = ? AND team_users.user_id = ?", app_id, user.ID).Scan(&app)
```

File: pkg/api/apps/logs.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	if result.Error != nil {
+		c.JSON(404, gin.H{"status": "error", "message": result.Error.Error()})
```
Comment (anirudhb) [inline pkg/api/apps/logs.go]: Should return 500 for server error

File: pkg/api/apps/logs.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+		return
+	} else if result.RowsAffected < 1 {
+		c.JSON(404, gin.H{"status": "error", "message": "App not found"})
```
Comment (anirudhb) [inline pkg/api/apps/logs.go]: Should return 400 for bad client request

File: pkg/api/apps/logs.go [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+		c.JSON(500, gin.H{"status": "error", "message": "Error creating Docker client"})
+		return
+	}
+
+	// Open a log stream via the Docker API
+	log_stream, err := cli.ContainerLogs(c.Request.Context(), cid, types.ContainerLogsOptions{
+		ShowStdout: true,
+		ShowStderr: true,
+		Follow:     true,
+		Tail:       "30",
```
Comment (anirudhb) [inline pkg/api/apps/logs.go]: Preferably make this configurable through a query parameter

**Review by anirudhb — CHANGES_REQUESTED** [review]
lgtm, nits and nice-to-haves

File: pkg/api/apps/logs.go (hunk 4 above)
Comment (cjdenio) [reply pkg/api/apps/logs.go]: Seems to me like 404 would be more representative of the actual error.

File: dokku_plugin/subcommands/cid/cid.go (hunk 1 above)
Comment (cjdenio) [reply dokku_plugin/subcommands/cid/cid.go:26]: Sounds good; we'll have to discuss whether or not we want to support multi-container apps in the future.

File: pkg/api/apps/logs.go (hunk 4 above)
Comment (anirudhb) [reply pkg/api/apps/logs.go]: Hmm, seems inconsistent with the rest of the API. Mind changing the other parts of the API then?

**Review by anirudhb — APPROVED** [review]
lgtm

**Pull request merged**